pkg/provider/helm/values.go

```diff
@@ -4,171 +4,430 @@
+				s.agentConfig.NodeAttestor: map[string]any{
+					"enabled": s.agentConfig.NodeAttestorEnabled,
+				},
+			},
+			"server": map[string]any{
+				"address": s.spireServerAddress,
+			},
+			"workloadAttestors": map[string]any{
+				s.agentConfig.WorkloadAttestor: s.agentConfig.WorkloadAttestorConfig,
+			},
+		},
+	}, nil
+}
+
+// generateValues generates the spire-server Helm values map.
+func (s *spireServerValues) generateValues() (map[string]any, error) {
+	if s.caKeyType == "" {
+		return nil, fmt.Errorf("caKeyType value is empty")
+	}
+
+	if s.caTTL == "" {
+		return nil, fmt.Errorf("caTTL value is empty")
+	}
+
+	if s.fullnameOverride == "" {
+		return nil, fmt.Errorf("fullnameOverride value is empty")
+	}
+
+	if s.logLevel == "" {
+		return nil, fmt.Errorf("logLevel value is empty")
+	}
+
+	if s.serverConfig.NodeAttestor == "" {
+		return nil, fmt.Errorf("serverConfig.NodeAttestor value is empty")
+	}
+
+	if s.serverConfig.NodeAttestorConfig == nil {
+		return nil, fmt.Errorf("serverConfig.NodeAttestorConfig value is nil")
+	}
+
+	if len(s.serverConfig.NodeAttestorConfig) == 0 {
+		return nil, fmt.Errorf("serverConfig.NodeAttestorConfig value is empty")
+	}
+
+	if s.serviceType == "" {
+		return nil, fmt.Errorf("serviceType value is empty")
+	}
+
+	return map[string]any{
+		"spire-server": map[string]any{
+			"caKeyType": s.caKeyType,
+			"caTTL":     s.caTTL,
+			"controllerManager": map[string]any{
+				"enabled": s.controllerManagerEnabled,
+			},
+			"fullnameOverride": s.fullnameOverride,
+			"logLevel":         s.logLevel,
+			"nodeAttestor": map[string]any{
+				s.serverConfig.NodeAttestor: s.serverConfig.NodeAttestorConfig,
+			},
+			"service": map[string]any{
+				"type": s.serviceType,
+			},
+		},
+	}, nil
+}
+
+// generateValues generates the spiffe-oidc-discovery-provider Helm values map.
+func (s *spiffeOIDCDiscoveryProviderValues) generateValues() (map[string]any, error) {
+	return map[string]any{
+		"spiffe-oidc-discovery-provider": map[string]any{
+			"enabled": s.enabled,
+		},
+	}, nil
+}
+
+// generateValues generates the spiffe-csi-driver Helm values map.
+func (s *spiffeCSIDriverValues) generateValues() (map[string]any, error) {
+	if s.fullnameOverride == "" {
+		return nil, fmt.Errorf("fullnameOverride value is empty")
+	}
+
+	return map[string]any{
+		"spiffe-csi-driver": map[string]any{
+			"fullnameOverride": s.fullnameOverride,
+		},
+	}, nil
+}
+
+// getNestedMap retrieves a nested map[string]any from a parent map.
+func getNestedMap(m map[string]any, key string) (map[string]any, bool) {
+	value, exists := m[key]
+	if !exists {
+		return nil, false
+	}
+
+	nestedMap := value.(map[string]any)
+	return nestedMap, true
+}
+
+// mergeMaps merges the source map into the destination map, returning a new merged map.
+func mergeMaps(src, dest map[string]any, overwriteExistingKeys bool) map[string]any {
+	merged := make(map[string]any)
+
+	for key, value := range dest {
+		merged[key] = value
```

Comment:
Worth noting that this refers to the value in `dest` for pointer types (affecting slices and maps). It might be worth just being explicit about that and modifying dest in place rather than trying to do a full deep copy?